backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/handlingunits/rest_api/move_hu/MoveHUCommand.java

```diff
@@ -51,17 +59,44 @@ private void executeInTrx()
 		{
 			final LocatorQRCode locatorQRCode = LocatorQRCode.ofGlobalQRCode(targetQRCode);
 
-			final HuId topLevelHUId = huTransformService.extractToTopLevelByQRCode(huId, huQRCode);
-
 			huMovementBL.moveHUs(HUMovementGenerateRequest.builder()
-					.toLocatorId(locatorQRCode.getLocatorId())
-					.huIdToMove(topLevelHUId)
-					.movementDate(SystemTime.asInstant())
-					.build());
+										 .toLocatorId(locatorQRCode.getLocatorId())
+										 .huIdsToMove(extractHUIdsToMove())
+										 .movementDate(SystemTime.asInstant())
+										 .build());
+		}
+		else if (HUQRCode.isTypeMatching(targetQRCode))
+		{
+			getMoveToTargetHUConsumer(HUQRCode.fromGlobalQRCode(targetQRCode))
+					.accept(extractHUIdsToMove());
 		}
 		else
 		{
 			throw new AdempiereException("Move target not handled: " + targetQRCode);
 		}
 	}
+
+	@NonNull
+	private Consumer<List<HuId>> getMoveToTargetHUConsumer(@NonNull final HUQRCode huqrCode)
```

Comment:
what's the benefit of returning Consumer? 
wondering why not:
1. inline it, similar to the code which is moving to locator
2. or change it to something like moveToTargetHU.... and the other branch to be extracted to a method called something like moveToTargetLocator

Reply:
bc this way it's easier to extend, when at some point we would like to extend to different HU target, like a TU or a CU